### PR TITLE
Add Unit tests

### PR DESCRIPTION
**Pull request by Cisco-the-wayword**
In this pull request, we implemented the following:
- Created a New folder spec
- Created test files for Person, Student, Teacher, Book, Classroom, Rental and decorator.rb file

### CODE REVIEW

**Review by emmanuellekamwa — CHANGES_REQUESTED** [review]
**STATUS: INVALID ⛔❌😢**
Hello @Cisco-the-wayword and @naanmohammed 👋
While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project where only one person has contributed. Please make sure everyone contributes equally to the project and submit again.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      expect(person.name).to eq(name)
+      expect(person.rentals).to be_empty
+    end
+  end
+
+  describe 'correct name' do
+    it 'returns persons name' do
+      expect(person.correct_name).to eq(name)
+    end
+  end
```
Comment (BenMukebo) [inline spec/person_spec.rb]: - [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure the `can_use_services?`, `of_age?`, and  `add_rental` class person method has (at least) an 'it' block 👇🏻 check your person.rb file

```ruby 
  def can_use_services?
    of_age? || @parent_permission
  end

  private

  def of_age
    @age.to_i >= 18
  end

  def add_rental(rental)
    @rentals.push(rental)
    rental.person = self
  end
```

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      expect(student.name).to eq(name)
+      expect(student.rentals).to be_empty
+    end
+  end
+
+  describe 'play hooky' do
+    it 'student plays hooky' do
+      expect(student.play_hooky).to eq('¯\\(ツ)/¯')
+    end
+  end
```
Comment (BenMukebo) [inline spec/student_spec.rb]: - [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure the `Student` class person method has (at least) an 'it' block 👇🏻 check your student.rb file

```ruby 
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @Cisco-the-wayword, @Mylo16,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- There is one file per class. ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL]  Please let's make sure that contributions to a group project are distributed equally among the members (commit messages should be fairly equal). Actually I can only see a single commit from @Mylo16, Kindly make sure to add more commit buy addressig the requiements. 👍🏻 


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Cisco-the-wayword, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title and summary. ✔️
- Add test ✔️
- Good use of Github flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**